File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.py
def pick(arr):
    for idx, a in enumerate(arr):
        if a > 0:
            return idx
    return -1
        
def solution(board, moves):
    answer = 0
    basket = []
    n = len(board)
    lines = {}
    l = 1
    for i in range(n):
        tmp = []
        for j in range(n):
            tmp.append(board[j][i])
        lines[l] = tmp
        l += 1
    
    for m in moves:
        pick_idx = pick(lines[m])
        if pick_idx >= 0: #뽑을수있음
            basket.append(lines[m][pick_idx])
            lines[m][pick_idx] = 0
        else:
            continue
        while len(basket) >= 2 and basket[-1] == basket[-2]:
            basket = basket[:-2]
            answer += 2
    return answer

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.swift
import Foundation

func solution(_ board:[[Int]], _ moves:[Int]) -> Int {
    
    var answer = 0
    let n = board.count
    var board = board
    
    var st = [Int]()
    for move in moves {
        var r = 0
        while r < n && board[r][move-1] == 0 {
            r += 1
        }
        if r == n { continue }
        
        let toy = board[r][move-1]
        if !st.isEmpty && st.last! == toy {
            st.removeLast()
            answer += 2
        } else {
            st.append(toy)
        }
        board[r][move-1] = 0
    }
    return answer
}